bin/fba_cmx

```diff
@@ -0,0 +1,817 @@
+    ('DESI_TARGET', '>i8'), ('BGS_TARGET', '>i8'), ('MWS_TARGET', '>i8'),
+    ('SCND_TARGET', '>i8'), ('TARGETID', '>i8'),
+    ('SUBPRIORITY', '>f8'), ('OBSCONDITIONS', 'i4'),
+    ('PRIORITY_INIT', '>i8'), ('NUMOBS_INIT', '>i8'), ('PRIORITY', '>i8'),
+    ('NUMOBS', '>i8'), ('NUMOBS_MORE', '>i8'), ('Z', '>f8'), ('ZWARN', '>i8'),
+    ('TIMESTAMP', 'S19'), ('VERSION', 'S14'), ('TARGET_STATE', 'S15')
+    ])
+
+
+
+# AR extra-hdu for dithering
+extradatamodel = np.array([], dtype=[
+    ('UNDITHER_RA','>f8'),('UNDITHER_DEC','>f8'),('TARGETID', '>i8')
+    ])
+
+
+
+# AR obscon
+if   (args.obscon=='dark,bright'): obscon = 'DARK|GRAY|BRIGHT'
+elif (args.obscon=='dark'):        obscon = 'DARK|GRAY'
+else:                              obscon = 'BRIGHT' # AR bright
+log.info('{:.1f}s\tsetting obscon = {}'.format(time()-start,obscon))
+
+
+
+# AR get matching index for two np arrays, those should be arrays with unique values, like id
+# AR https://stackoverflow.com/questions/32653441/find-indices-of-common-values-in-two-arrays
+# AR we get: A[maskA] = B[maskB]
+def unq_searchsorted(A,B):
+    # AR sorting A,B
+    tmpA  = np.sort(A)
+    tmpB  = np.sort(B)
+    # AR create mask equivalent to np.in1d(A,B) and np.in1d(B,A) for unique elements
+    maskA = (np.searchsorted(tmpB,tmpA,'right') - np.searchsorted(tmpB,tmpA,'left'))==1
+    maskB = (np.searchsorted(tmpA,tmpB,'right') - np.searchsorted(tmpA,tmpB,'left'))==1
+    # AR to get back to original indexes
+    return np.argsort(A)[maskA],np.argsort(B)[maskB]
+
+
+
+# AR ! not using make_mtl !
+# AR for commissioning, Adam says we should not use make_mtl, assign mtl columns by hand [email Oct, 17 2020]
+# AR by default, we propagate {PRIORITY,NUMOBS}_INIT to {PRIORITY,NUMOBS_MORE}
+# AR mtl (reproducing steps of make_mtl())
+def cmx_make_mtl(d,outfn):
+    # d     : output of read_targets_in_tiles()
+    # outfn : written fits file
+    mtl                = Table(d)
+    mtl.meta['EXTNAME']= 'MTL'
+    for col in ['NUMOBS_MORE', 'NUMOBS', 'Z', 'ZWARN', 'TARGET_STATE', 'TIMESTAMP', 'VERSION']:
+        mtl[col] = np.empty(len(mtl), dtype=mtldatamodel[col].dtype)
+    mtl['NUMOBS_MORE'] = mtl['NUMOBS_INIT']
+    mtl['PRIORITY']    = mtl['PRIORITY_INIT']
+    mtl['TARGET_STATE']= 'UNOBS'
+    mtl['TIMESTAMP']   = datetime.utcnow().isoformat(timespec='seconds')
+    mtl['VERSION']     = fiberassign.__version__            
+    obsconmask = set_obsconditions(d)    # AR : TBD : do we want to set obsconmask to 1? (see Ted s email)
+    mtl['OBSCONDITIONS'] = obsconmask
+    n,tmpfn = write_mtl(args.outdir,mtl.as_array(),indir=args.outdir,survey='cmx',ecsv=False)
+    if n:
+        os.rename(tmpfn,outfn)
+        log.info('{:.1f}s\tmtl targets written to {} , moved to {}'.format(time()-start,tmpfn,outfn)) 
+    else:
+        log.info('{:.1f}s\tmtl targets NOT written to {} (0 targets to write)'.format(time()-start,tmpfn)) 
+    return True
+
+
+
+# AR tiles
+if (dotile==True):
```

Comment:
This is redundant:  dotile is already a bool, so `if(dotile)` is fine.  Same with other cases below.